pkg/converter/converter_test.go

```diff
@@ -0,0 +1,196 @@
+		},
+		{
+			name: "complex",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:
+  components:
+    base:
+      enabled: true
+    pilot:
+      enabled: false
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: true`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+      enabled: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: "true"
+      enabled: false
+  namespace: %s
+  version: %s`, controlPlaneNamespace, istioVersion),
+			converterArgs: fmt.Sprintf("%s %s -v %s -n %s", istioFile, sailFile, istioVersion, controlPlaneNamespace),
+		},
+		{
+			name: "mandatory-arguments-only",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: %s`, controlPlaneNamespace),
+			converterArgs: istioFile,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Cleanup(func() {
+				os.Remove(istioFile)
+				os.Remove(sailFile)
+			})
+			g := NewWithT(t)
+			err := saveYamlToFile(tc.input, istioFile)
+			g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+			g.Expect(executeConfigConverter(tc.converterArgs)).To(Succeed(), "error in execution of ./configuration-converter.sh")
+
+			isConversionValidated, err := validateYamlContent(sailFile, tc.expectedOutput)
+			g.Expect(err).To(Succeed(), fmt.Errorf("Error on validation: %w", err))
+			g.Expect(isConversionValidated).To(BeTrue(), "Converted content is not as expected")
+		})
+	}
+}
+
+// saveYamlToFile writes the given YAML content to given file
+func saveYamlToFile(input, istioFile string) error {
+	if err := os.WriteFile(istioFile, []byte(input), 0o644); err != nil {
+		return fmt.Errorf("failed to write YAML file: %w", err)
+	}
+	return nil
+}
+
+func executeConfigConverter(scriptArgs string) error {
+	converter := filepath.Join(project.RootDir, "tools", "configuration-converter.sh")
+
+	cmdArgs := []string{converter, scriptArgs}
+
+	cmd := strings.Join(cmdArgs, " ")
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		fmt.Printf("Script output: %s", output)
+		return fmt.Errorf("error in execution of command %v: %w", cmdArgs, err)
+	}
+	return nil
+}
+
+// compareYamlContent checks if the provided YAML content matches the content of converted sail operator config
+func validateYamlContent(sailFile, expectedOutput string) (bool, error) {
```

Comment:
I would actually get rid of this function and just inline the code in the test itself. If you replace all the ifs with g.Expect(), this would end up being just 3 lines in the test function.